processing/src/main/java/org/apache/druid/segment/column/StringDictionaryEncodedColumn.java

```diff
@@ -121,7 +121,11 @@ public HistoricalDimensionSelector makeDimensionSelector(
       @Override
       public int getValueCardinality()
       {
-        return getCardinality();
+        if (extractionFn != null) {
+          return CARDINALITY_UNKNOWN;
```

Comment:
Should this be based on `ExtractionType` since we still know the cardinality if it is `ONE_TO_ONE`, or is that not reliably set on all extractionFn implementations?

Reply:
@clintropolis thanks for checking.

`ONE_TO_ONE` isn't enough as we also need to honor the ordering (see update to `getValueCardinality()` javadoc, but I realized that `ExtractionFn` has a `preservesOrdering()` method as well ... if both of them are true then we can say cardinality is same.

even after that change, I see many of the tests fail for GroupBy-V1 ... it was working before incorrectly. So, I see that this PR might be problematic for users who had couple of queries working and they stopped working suddenly. so I am removing the "fix" as it causes more nuisance than it fixes. I am leaving only the documentation change in this PR.

I think better fix would be to use a `ExtractionFnVirtualColumn` and remove `DimensionSpec.getExtractionFn()` altogether. ( related https://github.com/apache/incubator-druid/issues/8242 )